src/workers/parser/mapAwaitExpression.js

```diff
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+// @flow
+
+import { parse } from "./utils/ast";
+import generate from "@babel/generator";
+import * as t from "@babel/types";
+
+function hasTopLevelAwait(expression: string) {
+  const ast = parse(expression, { allowAwaitOutsideFunction: true });
+  const hasAwait = ast.program.body.some(
```

Comment:
Seems like this should be a traversal, since the `await` can also be in nested expressions somewhere, as long as it's not in some other nested async function.